corehq/form_processor/backends/sql/ledger.py

```diff
@@ -77,6 +77,7 @@ def _lazy_original_balance():
         ledger_value.track_create(transaction)
         # only do this after we've created the transaction otherwise we'll get the wrong delta
         ledger_value.balance = new_ledger_values.balance
+        ledger_value.last_modified = stock_trans.timestamp
```

Comment:
One downside with using server times for ledger dates is that users can only submit stock updates in real-time.  For instance, I worked on a project where the mobile workers would go around at the end of each month and record the EOD totals for each day.